karate-core/src/main/java/com/intuit/karate/core/ScenarioContext.java

```diff
@@ -240,9 +241,20 @@ public InputStream getResourceAsStream(String name) {
             info.put("scenarioName", unit.scenario.getName());
             info.put("scenarioDescription", unit.scenario.getDescription());
             info.put("scenarioType", unit.scenario.getKeyword());
+            info.put("scenarioExampleIndex", unit.scenario.getExampleIndex());
```

Comment:
can you try one more thing, because I'd like to avoid adding more to `karate.info` - since we now have `scenario` part of the context. there are a few things like `scenario.index` that deserve to be added

suppose we make the `scenario` variable private and have a getter that switches between `unit.scenario` or `context.scenario` then you can do this:

`karate.context.scenario.exampleIndex`

I'm also okay to have a `scenario` getter on the `ScriptBridge` to make it more concise: `karate.scenario.exampleIndex`

then in the long term we can deprecate `karate.info`.

WDYT ?

Reply:
agreed.  can you clarify.  did you want my property additions to karate.info reverted and gotten via a new call at the ScriptBridge?

Reply:
yes. if you can get all the data you need via ScriptBridge.getScenario() - lets not add any more to karate.info - so it can be deprecated / discouraged in future.

Reply:
otes --
let me know what you think.
rolled back all code touching 'info()' to let that live it's final days to be sunsetted in isolation.
'scenario' marked private and created smart getter to basically do a little of what 'getScenarioInfo()' and 'info()' do to arrive at a Scenario object.
and moved all work on the Scenario object to display data to the ScriptBridge.

usage --
karate-config.js
karate.log(karate.getScenario());

log output for a scenario
{"timestamp":"2020-08-10 12:33:48.295","level":"INFO","thread":"ForkJoinPool-1-worker-3","logger":"com.intuit.karate","message":"{\n  \"scenarioExampleData\": null,\n  \"scenarioType\": \"Scenario\",\n  \"scenarioDescription\": \"\",\n  \"scenarioName\": \"order 1\",\n  \"scenarioIndex\": -1,\n  \"scenarioExampleIndex\": -1\n}\n ","context":"default"}

log output for a scenario outline
{"timestamp":"2020-08-10 12:30:33.766","level":"INFO","thread":"ForkJoinPool-1-worker-7","logger":"com.intuit.karate","message":"{\n  \"scenarioExampleData\": {\n    \"data1\": \"d111\",\n    \"data2\": \"d222\"\n  },\n  \"scenarioType\": \"Scenario Outline\",\n  \"scenarioDescription\": \"\",\n  \"scenarioName\": \"order 4\",\n  \"scenarioIndex\": 2,\n  \"scenarioExampleIndex\": 2\n}\n ","context":"default"}
{"timestamp":"2020-08-10 12:30:33.780","level":"INFO","thread":"ForkJoinPool-1-worker-5","logger":"com.intuit.karate","message":"{\n  \"scenarioExampleData\": {\n    \"data1\": \"d11\",\n    \"data2\": \"d22\"\n  },\n  \"scenarioType\": \"Scenario Outline\",\n  \"scenarioDescription\": \"\",\n  \"scenarioName\": \"order 4\",\n  \"scenarioIndex\": 1,\n  \"scenarioExampleIndex\": 1\n}\n ","context":"default"}
{"timestamp":"2020-08-10 12:30:33.771","level":"INFO","thread":"ForkJoinPool-1-worker-3","logger":"com.intuit.karate","message":"{\n  \"scenarioExampleData\": {\n    \"data1\": \"d1\",\n    \"data2\": \"d2\"\n  },\n  \"scenarioType\": \"Scenario Outline\",\n  \"scenarioDescription\": \"\",\n  \"scenarioName\": \"order 4\",\n  \"scenarioIndex\": 0,\n  \"scenarioExampleIndex\": 0\n}\n ","context":"default"}

Reply:
@ptrthomas one refinement i'm thinking about as i've been using a private build of this today.  what do you think of shortening the property names as the "getScenario" is pretty descriptive?  i initially kept the property names the same to stay in alignment with what's already in place.

what is there now from my changes
            info.put("scenarioName", scenario.getName());
            info.put("scenarioDescription", scenario.getDescription());
            info.put("scenarioType", scenario.getKeyword());
            info.put("scenarioIndex", scenario.getIndex());
            info.put("scenarioExampleIndex", scenario.getExampleIndex());
            info.put("scenarioExampleData", scenario.getExampleData());

what i'm thinking of for a refinement
            info.put("name", scenario.getName());
            info.put("description", scenario.getDescription());
            info.put("type", scenario.getKeyword());
            info.put("index", scenario.getIndex());
            info.put("exampleIndex", scenario.getExampleIndex());
            info.put("exampleData", scenario.getExampleData());